src/state/hooks.ts

```diff
@@ -30,8 +38,12 @@ export const useAppDispatch = (): AppDispatch => {
  */
 export function useAppSelector<T>(selector: (state: RechartsRootState) => T): T | undefined {
   const context = useContext(ReactReduxContext);
-  if (context) {
```

Comment:
This is an interesting one.

The old code was not enough to re-run the selector on state change; the Redux context itself never actually changes, so React was never calling this hook for the second time.

The right way is to use store.subscribe and store.getState() which conveniently is precisely what `useSyncExternalStoreWithSelector` does. I used the shim to allow us to continue supporting React 16. After we decide to support only React 18+ we can import that from `react` directly.

Reply:
very interesting.. 
I don't have great knowledge of this hook and how it works